### DISCUSSION TITLE
GitHub Automatic Deployment Not Setting Environment Variables

### DISCUSSION THREAD

**Discussion by washubaseball** [vercel/community#5347] [Help] [answered]
### Summary

Hello,
I have been having an issue with automatic deployments from GitHub. I have a NEXT_PUBLIC_API_URL environment variable set in Vercel (one for a preview environment and one for a production environment). When I push a change to either branch (main or test), the automatic deployment runs, but the environment variables are not set (my app will attempt to fetch something from my API but it won't be able to since the environment variable is not set). When I redeploy the branch (without cache) manually, it sets the environment variable fine. I have just been doing this over the last year as a workaround, but obviously this is quite annoying. Any help would be appreciated! I tried setting VERCEL_FORCE_NO_BUILD_CACHE=1 but that still did not work. 

### Steps to Reproduce

I have an NextJS app within an Nx workspace, with a build step of "npx nx build boost-web --prod" with the output directory being: "dist/apps/boost-web/.next". I have an environment variable "NEXT_PUBLIC_API_URL" set within Vercel's web application for each branch (main and test). When you push code to either branch, it will deploy successfully, but the environment variable will not be set. When you click the dropdown menu on that deployment (and click redeploy without build cache), it will set the environment variable fine.

**Comment by davidwinter** [comment]
@washubaseball I've actually started to experience this only this morning. Had been working fine until then. Environment variables aren't being detected which I have set within Vercel, and they haven't been modified in over half a year 🤔 

I tried redeploying manually from the Vercel web app, but that didn't seem to change anything for me.

**Comment by typesafeui** [comment]
I have the same issue too 🥲

**Comment by davidwinter** [comment]
I've submitted a support request, so fingers crossed it will be looked into soon. I made reference to this discussion.

**Reply by typesafeui** [reply]
finger crossed!

**Comment by MarkZakelj** [comment]
Same thing happening to me!

**Comment by viktorlarsson** [comment]
Same thing here.

**Comment by andrewdoro** [comment]
I can't deploy, fuck it

**Comment by davidwinter** [comment]
The status page only seems to refer to an issue with Speed Insights. Seems like this is affecting a fair few people in this thread, so surprised it's not been picked up yet as I would assume it's a bit more widespread:

https://www.vercel-status.com/

**Reply by typesafeui** [reply]
yep not a good sign that the status page has not picked it up yet 💀

**Reply by GRA0007** [reply]
Looks like it's on there now!

**Comment by rodriguezaitor** [comment]
same here

**Comment by GRA0007** [comment]
Thought I was going crazy, everything works locally, just missing some environment variables when I deploy on Vercel.

**Comment by julamb** [comment]
Same here! 

Could not make it work at all, even with redeploying manually.

**Comment by anirudhsama** [comment]
Same here!

**Comment by bbirec** [comment]
Same here.

There's something suspicious about the serverless function's location. I have been using the default location (ida1), but the serverless logs indicate that the new deployments are actually being placed in Tokyo, Japan (hnd1).

**Reply by GRA0007** [reply]
Strange, my function region hasn't changed

**Comment by Heljas** [comment]
Same here, looks like our functions are running in arn1 instead of fra1.

**Comment by k9ordon** [comment]
It has now been mentioned on Vercel's status page as well. https://www.vercel-status.com/incidents/snybsxv4v0vc

**Reply by typesafeui** [reply]
Nice ✨

**Comment by typesafeui** [comment, marked as answer]
A fix has been implemented 👀 : https://www.vercel-status.com/incidents/snybsxv4v0vc

**Comment by typesafeui** [comment]
Can confirm that it works for me now😊

**Comment by imranbarbhuiya** [comment]
I'm having this error today

**Comment by ytwater** [comment]
I'm also having this problem.  I tried removing the variables from the Environmental Variables Tab of my dashboard and re-adding them.  I trued using the vercel cli to remove and add.  I've tried preview variables, and preview variables tied to a specific branch.

a `console.log(process.env)` in the build process shows that some of the variables I've set in the dashboard are being sent, others are not.

**Reply by ytwater** [reply]
I opened an issue with Vercel and they solved my issue.  I had updated from turbo v1 to turbo v2, and had not been updating my turbo.json with my environmental variables.  I added all my variables into turbo.json and it's back and working!

------ from support -------

"The upgrade from Turbo v1 to v2 changed the way that variables were handled by default. In v1, it used Loose mode, where variables didn't need to be defined in your `turbo.json`. However, v2 uses Strict mode, meaning that all variables need to be defined, and if they aren't, the build step will no longer include these when building your project."

"Please see here for more information: https://turbo.build/repo/docs/crafting-your-repository/upgrading#update-turbo-run-commands"

**Reply by cangel111** [reply]
Thanks - this worked for me <3

[a comment by an automated account is not shown]
